containers/product/index.js

```diff
@@ -18,12 +19,47 @@ export default function Product ({ slug }) {
 
   const { productBySlug } = data
 
+  const hasVariants = () => productBySlug.variants.nodes.length > 0
+  const defaultVariantId = () =>
+    hasVariants()
+      ? productBySlug.variants.nodes[0].id
+      : productBySlug.masterVariant.id
+
   return (
     <div>
       <h1>{productBySlug.name}</h1>
-      <button onClick={() => {
-        addToCart({ variables: { variantId: productBySlug.masterVariant.id, quantity: 1 } });
-      }}>Add to cart</button>
+      <Formik
+        initialValues={{
+          quantity: 1,
+          variantId: defaultVariantId()
+        }}
+        onSubmit={async values => {
+          addToCart({
+            variables: {
+              variantId: values.variantId,
+              quantity: values.quantity
+            }
+          })
+        }}
+      >
+        <Form>
+          {productBySlug.variants.nodes.map(variant => (
+            <div key={variant.id}>
+              <label>
+                <Field type='radio' name='variantId' value={variant.id} />
+                {variant.sku}
+              </label>
+              <br />
```

Comment:
I'd avoid using `br` tags. Also, I don't believe it's needed here.

Reply:
👍